calltable.cpp

```diff
@@ -1959,8 +1959,10 @@ Call::_save_rtp(packet_s *packetS, char is_fax, char enable_save_packet, bool re
 	}
 	if(enable_save_packet) {
 		if((this->silencerecording || (this->flags & FLAG_SAVERTPHEADER)) && !this->isfax && !record_dtmf) {
-			if(packetS->datalen_() >= RTP_FIXED_HEADERLEN &&
-			   packetS->header_pt->caplen > (unsigned)(packetS->datalen_() - RTP_FIXED_HEADERLEN)) {
+			if(packetS->isStun() || packetS->isDtls()) {
+				save_packet(this, packetS, TYPE_RTP, forceVirtualUdp);
+			} else if(packetS->datalen_() >= RTP_FIXED_HEADERLEN &&
+			          packetS->header_pt->caplen > (unsigned)(packetS->datalen_() - RTP_FIXED_HEADERLEN)) {
 				unsigned int tmp_u32 = packetS->header_pt->caplen;
 				packetS->header_pt->caplen = min(packetS->header_pt->caplen - (packetS->datalen_() - RTP_FIXED_HEADERLEN),
 								 packetS->dataoffset_() + RTP_FIXED_HEADERLEN);
```

Comment:
I have moved STUN and DTLS checks inside the `savertp = header` case, so this makes sure that `savertp = yes` case conditions are not altered. Also `savertp = no` does not even reach this flow.